modules/core/src/main/java/org/apache/ignite/internal/util/distributed/FullMessage.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.util.distributed;
+
+import java.io.Serializable;
+import java.util.Map;
+import java.util.UUID;
+import org.apache.ignite.internal.managers.discovery.DiscoCache;
+import org.apache.ignite.internal.managers.discovery.DiscoveryCustomMessage;
+import org.apache.ignite.internal.managers.discovery.GridDiscoveryManager;
+import org.apache.ignite.internal.processors.affinity.AffinityTopologyVersion;
+import org.apache.ignite.internal.util.typedef.internal.S;
+import org.apache.ignite.lang.IgniteUuid;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Full process message. Contains single nodes results.
+ *
+ * @param <R> Result type.
+  */
+public class FullMessage<R extends Serializable> implements DiscoveryCustomMessage {
+    /** Serial version uid. */
+    private static final long serialVersionUID = 0L;
+
+    /** Custom message ID. */
+    private final IgniteUuid id = IgniteUuid.randomUuid();
+
+    /** Process id. */
+    private final UUID processId;
+
+    /** Process type. */
+    private final int type;
+
+    /** Result. */
+    private Map<UUID, R> res;
+
+    /** Error. */
+    private Map<UUID, Exception> err;
+
+    /**
+     * @param processId Process id.
+     * @param type Process type.
+     * @param res Results.
+     * @param err Errors
+     */
+    public FullMessage(UUID processId, int type, Map<UUID, R> res, Map<UUID, Exception> err) {
```

Comment:
It seems we should change `type` from int to the `DistributedProcessType` and use type.ordinal() if required.